ConsoleApp/Stack.cs

```diff
@@ -0,0 +1,67 @@
+namespace ConsoleApp
+{
+    public class Stack<T>
+    {
+        private class Element
+        {
+            public T? Value { get; set; }
+            public Element? Next { get; set; }
+        }
+
+        private Element? _top;
+        public int Count { get; private set; }
+
+        public void Push(T value)
+        {
+            Element top = new Element
+            {
+                Value = value
+            };
+
+            top.Next = _top;
+            _top = top;
+            Count++;
+        }
+
+        public T? Pop()
+        {
+            if (_top != null)
+            {
+                Element? top = _top;
+                _top = _top.Next;
+                Count--;
+                
+                return top.Value;
+            }
+
+            return default(T);
```

Comment:
that's wrong behavior that can create a false expectations that the stack is not empty, an exception is the only right decision here